src/Apps/Order/Components/AddressModal.tsx

```diff
@@ -187,65 +208,28 @@ export const AddressModal: React.FC<Props> = ({
               )}
               <AddressModalFields />
               <Spacer mb={2} />
-              {user && !userHasLabFeature(user, "Phone Number Validation") && (
-                <Input
-                  title="Phone number"
-                  description="Required for shipping logistics"
-                  placeholder="Add phone number"
-                  name="phoneNumber"
-                  type="tel"
-                  onChange={formik.handleChange}
-                  onBlur={formik.handleBlur}
-                  error={
-                    formik.touched.phoneNumber && formik.errors.phoneNumber
+
+              <PhoneNumberInput
+                mt={4}
+                phoneNumber={formik.values?.phone}
+                onChange={async (region, number) => {
+                  if (region && number && relayEnvironment) {
+                    const phoneInformation = await getPhoneNumberInformation(
+                      number,
+                      relayEnvironment,
+                      region
+                    )
```

Comment:
I agree a call for every onChange event is a bit excessive.

Reply:
I haven't experienced slowness but yeah I have speedy wifi :) 

if we debounce though, we will need to figure out when to validate the number. when user types in the last digit of their phone number, we need to make a call to get it validated so that we enable `Save` button that depends on `isValid` property. and it's (almost?) impossible to know when they type the last digit to decide whether or not to debounce.